### PR TITLE
fix(Script/Mandorik): enrage on reset and on subsequent pulls.

### PR DESCRIPTION
**Pull request by ChrisCGalbraith**
## Changes Proposed:
-  Added checks to remove frenzy aura within Reset()
-  Correctly set Ohgans boss state to NOT_STARTED when Mandokir's Reset() is called.

## Issues Addressed:

- Closes part of: https://github.com/azerothcore/azerothcore-wotlk/issues/834 

## Tests Performed:

- Tested on Windows10 64-bit in-game following the steps outline below.
- Tested additional bug where if you had killed Ohgan then reset the fight, Ohgans data remained as DONE and mandokir
   would immediately enrage on subsequent pulls.

## How to Test the Changes:

1. Pull Mandokir and .damage or .die Ohgan, see Mandokir enrage.
2. Reset the fight by turning GM on or dying. Observe that enrage is removed on reset.
3. Additionally, observe that mandokir does not enrage if you have killed ohgan before him, and then re-pulled him.

## Known Issues and TODO List:

- [This PR is complete, however it does not resolve all the issues outlined in the linked issue #834 ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by Helias — APPROVED** [review]

File: src/server/scripts/EasternKingdoms/ZulGurub/boss_mandokir.cpp [hunk 1]
```diff
@@ -105,14 +105,17 @@ class boss_mandokir : public CreatureScript
...
             {
-                killCount = 0;
                 events.ScheduleEvent(EVENT_CHECK_START, 1000);
                 if (Creature* speaker = ObjectAccessor::GetCreature(*me, instance->GetGuidData(NPC_VILEBRANCH_SPEAKER)))
                     if (!speaker->IsAlive())
                         speaker->Respawn(true);
             }
+
+            killCount = 0;
+            me->RemoveAura(SPELL_FRENZY);
```
Comment (Kitzunu) [inline src/server/scripts/EasternKingdoms/ZulGurub/boss_mandokir.cpp:115]:
```suggestion
            me->RemoveAuraDueToSpell(SPELL_FRENZY);
```

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: src/server/scripts/EasternKingdoms/ZulGurub/boss_mandokir.cpp (hunk 1 above)
Comment (Kitzunu) [reply src/server/scripts/EasternKingdoms/ZulGurub/boss_mandokir.cpp:115]: Always use RemoveAurasDueToSpell when you use spell id

**Comment by T1ti** [comment]
isn't that just https://github.com/azerothcore/azerothcore-wotlk/issues/7153#issuecomment-889945944 ? 
it's a global issue

File: src/server/scripts/EasternKingdoms/ZulGurub/boss_mandokir.cpp (hunk 1 above)
Comment (ChrisCGalbraith) [reply src/server/scripts/EasternKingdoms/ZulGurub/boss_mandokir.cpp]: Good to know - cheers Kitzunu

**Comment by pklloveyou** [comment]
Tested

**Review by Kitzunu — APPROVED** [review]

**Comment by Kitzunu** [comment]
Thanks

**Pull request merged**